content/posts/2024-11-05-flang-new.md

```diff
@@ -0,0 +1,444 @@
+---
+author: "David Spickett"
+date: "2024-11-05"
+tags: ["Flang", "MLIR"]
+title: "LLVM Fortran Levels Up: Goodbye flang-new, hello flang!"
+---
+
+LLVM has included a Fortran compiler "Flang" since LLVM 11. What you might not
+know is that the name of Flang's binary was not `flang`, but `flang-new`.
+
+The LLVM 20 release brings an end to the era of `flang-new`. The
+community has decided Flang has reached a level of quality that is worthy of a
+new name.
+
+The "new" name? You guessed it, `flang`.
+
+In this article I will explain why this simple change actually represents a
+major milestone for the Flang project. Covering Flang's almost 10 year journey to
+into the LLVM Project and its adoption of one of LLVM's hottest new features, MLIR.
```

Comment:
It's 10 yrs if you consider "Flang" == "LLVM Flang" | "Classic Flang". I've been trying to make clear distinction to avoid confusion. Ultimately, these are two completely different projects. And some (most?) commercial are based on "Classic" rather than "LLVM" Flang.

Reply:
In this context I'm referring to the genesis of an open source LLVM based Fortran compiler that in some way got named flang. So 10 years since the National Labs announcement.

But I can be clearer here, if it didn't make sense to the reader then it needs to change regardless.